code/__HELPERS/text.dm

```diff
@@ -1233,3 +1237,50 @@ GLOBAL_LIST_INIT(binary, list("0","1"))
 	for(var/iteration in 1 to length_char(text))
 		grawlix += pick("@", "$", "?", "!", "#", "§", "*", "£", "%", "☠", "★", "☆", "¿", "⚡")
 	return grawlix
+
+/// Goes through the input and removes any punctuation from the start and end of the string.
+/proc/strip_punctuation(input)
+	// Makes sure " hey - " properly drops the hyphen
+	input = trim(input)
+
+	var/static/list/bad_punctuation = list("!", "?", ".", "~", ";", ":", "-", "|", "+", "_", ",", "`")
+	var/last_char = copytext_char(input, -1)
+	while(last_char in bad_punctuation)
+		input = copytext(input, 1, -1)
+		last_char = copytext_char(input, -1)
+
+	var/first_char = copytext_char(input, 1, 2)
+	while(first_char in bad_punctuation)
+		input = copytext(input, 2)
+		first_char = copytext_char(input, 1, 2)
+
+	// one last trim so we wend up with "hey"
+	input = trim(input)
+	return input
```

Comment:
As well as here. I won't even test this case. It's 100% will be better with regex